src/PagerViewAdapter.tsx

```diff
@@ -124,7 +125,7 @@ export default function PagerViewAdapter<T extends Route>({
         {...rest}
         ref={pagerRef}
         style={[styles.container, style]}
-        initialPage={index}
+        initialPage={initialIndexRef.current}
```

Comment:
If the `index` get changed before `<ViewPager>` mounted: since only the very initial index being kept and passed to pager now, and the followed `index` change update will be missed ( `pagerRef.current?.setPage(index)` in `useEffect()` does nothing due to `pagerRef` not received yet), it may leave the selected page in a wrong state.